lib/web/apiserver.go

```diff
@@ -3616,3 +3617,11 @@ func (h *Handler) authExportPublic(w http.ResponseWriter, r *http.Request, p htt
 	// It also handles the Range negotiation
 	http.ServeContent(w, r, "authorized_hosts.txt", time.Now(), reader)
 }
+
+// getIsDashboard returns a bool indicating if the cluster is a
+// dashboard cluster.
+// Dashboards are clusters running on cloud infrastructure that aren't
+// Teleport Cloud users
+func getIsDashboard(features proto.Features) bool {
+	return !features.GetCloud() && features.GetRecoveryCodes()
```

Comment:
Should we make a new flag `isRunningOnCloudInfrastructure` instead of checking if recovery codes are enabled? We can derive recovery codes from that new flag. Thoughts?

Reply:
"on cloud infrastructure" means on _our_ internal cloud infra, and not just any cloud right? I think that'd be a nice add

Reply:
> "on cloud infrastructure" means on our internal cloud infra, and not just any cloud right?

Correct